src/NuGet.Services.Metadata.Catalog.Monitoring/Resources/PackageRegistrationMetadataResourceV3.cs

```diff
@@ -26,31 +27,52 @@ public class PackageRegistrationMetadataResourceV3 : IPackageRegistrationMetadat
 
         public async Task<PackageRegistrationIndexMetadata> GetIndex(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            var feedPackage = await GetPackageFromIndex(package, log, token);
-            return feedPackage != null ? 
-                JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) : 
-                null;
+            try
+            {
+                var feedPackage = await GetPackageFromIndex(package, log, token);
+                return feedPackage != null ?
+                    JsonConvert.DeserializeObject<PackageRegistrationIndexMetadata>(feedPackage.ToString()) :
+                    null;
+            }
+            catch (Exception e)
+            {
+                throw new ValidationException($"Could not fetch {nameof(PackageRegistrationIndexMetadata)} from V3!", e);
+            }
         }
 
         public async Task<PackageRegistrationLeafMetadata> GetLeaf(PackageIdentity package, ILogger log, CancellationToken token)
         {
-            var feedPackage = await GetPackageFromLeaf(package, log, token);
-            return feedPackage != null ?
-                JsonConvert.DeserializeObject<PackageRegistrationLeafMetadata>(feedPackage.ToString()) :
-                null;
+            try
+            {
+                var feedPackage = await GetPackageFromLeaf(package, log, token);
+                return feedPackage != null ?
+                    JsonConvert.DeserializeObject<PackageRegistrationLeafMetadata>(feedPackage.ToString()) :
+                    null;
+            }
+            catch (Exception e)
+            {
```

Comment:
Ah I remember now--the try-catch in `GetPackageFromLeaf` is necessary because if we get a 404 we want to return null because the blob is missing. If we get a deserialization error then we want to throw and bubble up.